### PR TITLE
chore(weave): Use non-mutable defaults for clickhouse schema

### PR DESCRIPTION
**Pull request by andrewtruong**
## Summary
1. Don't use mutable defaults!

### CODE REVIEW

File: weave/trace_server/clickhouse_schema.py [hunk 1]
```diff
@@ -16,7 +16,9 @@ class CallStartCHInsertable(BaseModel):
     attributes_dump: str
     inputs_dump: str
     input_refs: typing.List[str]
-    output_refs: typing.List[str] = []  # sadly, this is required
+    output_refs: typing.List[str] = Field(
+        default_factory=list
+    )  # sadly, this is required
```
Comment (andrewtruong) [inline weave/trace_server/clickhouse_schema.py:21]: I'm not sure what this "is required" means.  What does it mean?

Comment (gtarpenning) [reply weave/trace_server/clickhouse_schema.py:21]: Inserting into the DB fails without `[]`, as the insertion does a concat with the existing list, so if its a `nonetype` the db throws

Comment (gtarpenning) [reply weave/trace_server/clickhouse_schema.py:21]: what does this object look like if you don't specify `output_refs` now?

Comment (andrewtruong) [reply weave/trace_server/clickhouse_schema.py:21]: Ok I think that's fine behaviour then, not sure why the "sadly" part

This has the same behaviour as before, just not using a mutable default.

Comment (andrewtruong) [reply weave/trace_server/clickhouse_schema.py:21]: Here's what it will look like:

image

**Review by gtarpenning — APPROVED** [review]

**Pull request merged**